src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Perf.Segment.cs

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Buffers;
+using System.Collections.Generic;
+
+namespace System.Text.Json.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public class Perf_Segment
+    {
+        // Keep the JsonStrings resource names in sync with TestCaseType enum values.
+        public enum TestCaseType
+        {
+            Json4KB,
+            Json40KB,
+            Json400KB
+        }
+
+        private string _jsonString;
+        private byte[] _dataUtf8;
+        private Dictionary<int, ReadOnlySequence<byte>> _sequences;
+        private ReadOnlySequence<byte> _sequenceSingle;
+
+        [ParamsSource(nameof(TestCaseValues))]
+        public TestCaseType TestCase;
+
+        public static IEnumerable<TestCaseType> TestCaseValues() => (IEnumerable<TestCaseType>)Enum.GetValues(typeof(TestCaseType));
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _jsonString = JsonStrings.ResourceManager.GetString(TestCase.ToString());
+
+            _dataUtf8 = Encoding.UTF8.GetBytes(_jsonString);
+
+            _sequenceSingle = new ReadOnlySequence<byte>(_dataUtf8);
+
+            int[] segmentSizes = { 4_096, 8_192 };
+
+            _sequences = new Dictionary<int, ReadOnlySequence<byte>>();
+
+            for (int i = 0; i < segmentSizes.Length; i++)
```

Comment:
nit: could be simplified

```cs
foreach (int segmentSize in new int[] { 4_096, 8_192 })
{
   _sequences.Add(segmentSize, Utf8JsonReaderCommentsTests.GetSequence(_dataUtf8, segmentSize));
}
```

or just:

```cs
 _sequences.Add(4_096, Utf8JsonReaderCommentsTests.GetSequence(_dataUtf8, 4_096));
 _sequences.Add(8_192 , Utf8JsonReaderCommentsTests.GetSequence(_dataUtf8, 8_192 ));
```

also, `4_096` and `8_192` are magic numbers, perhaps we should have two consts that should be resued?

```cs
const int smallSegmentSize = 4_096;
const int bigSegmentSize = smallSegmentSize * 2;
```